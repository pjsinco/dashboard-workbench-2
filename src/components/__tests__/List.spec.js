import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'
import List from '@/components/List.vue'

describe('List.vue', () => {
  it('renders li for each item in props.items', () => {
    const items = ['foo', 'bar', 'baz']
    const wrapper = shallow(List, { propsData: { items } })

    expect(wrapper.findAll('li')).toHaveLength(items.length)
  })

  it('has the same HTML structure', () => {
    const items = ['foo', 'bar', 'baz']
    const renderer = createRenderer()
    const wrapper = shallow(List, { propsData: { items } })
    
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})

