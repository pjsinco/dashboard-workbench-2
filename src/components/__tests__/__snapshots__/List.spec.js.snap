// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List.vue has same HTML structure 1`] = `
<ul>
    <li>item one</li>
    <li>item two</li>
</ul>
`;

exports[`List.vue has the same HTML structure 1`] = `
<ul>
    <li>foo</li>
    <li>bar</li>
    <li>baz</li>
</ul>
`;
